src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -321,6 +334,26 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                         $params = array_merge($params, $data[$paramName]);
                     }
                 } elseif ($allowed && !$ignored && (isset($data[$key]) || array_key_exists($key, $data))) {
+                    if ($this->propertyInfoExtractor) {
+                        /**
+                         * @var Type[]
```

Comment:
Although the called function has proper doc blocks the array cast makes IDEs confused about the type of `$types`.

Hence the comments for type-hinting to ensure the IDE supplies method completion for `$type`.